src/Catalog.API/Extensions/Extensions.cs

```diff
@@ -28,17 +29,19 @@ public static void AddApplicationServices(this IHostApplicationBuilder builder)
         builder.Services.AddOptions<CatalogOptions>()
             .BindConfiguration(nameof(CatalogOptions));
 
-        if (builder.Configuration["AI:Onnx:EmbeddingModelPath"] is string modelPath &&
-            builder.Configuration["AI:Onnx:EmbeddingVocabPath"] is string vocabPath)
+        if (builder.Configuration["AI:Ollama:EmbeddingName"] is string ollamaEmbeddingModel)
         {
-            builder.Services.AddBertOnnxTextEmbeddingGeneration(modelPath, vocabPath);
+            builder.Services.AddSingleton<IEmbeddingGenerator<string, Embedding<float>>>(new OllamaEmbeddingGenerator(new Uri("http://localhost:11434"), ollamaEmbeddingModel));
```

Comment:
What URL is this? Can we use service discovery or an injected address via configuration?

Reply:
This is Ollama's default url. I had 10 minutes and just hardcoded it. The ideal would be we'd have an Ollama resource that would actually stand it up locally, get its url, feed it in, etc., but to my knowledge we don't right now. Second best would be pulling it from config, with a default that matches Ollama's default.

Reply:
Do we need to support Ollama in eshop right now? We didn't support it before.